go/pkg/client/tree.go

```diff
@@ -190,21 +193,39 @@ func loadFiles(execRoot string, excl []*command.InputExclusion, path string, fs
 	}
 
 	if len(files) == 0 {
-		fs[normPath] = &fileSysNode{emptyDirectoryMarker: true}
+		fs.set(normPath, &fileSysNode{emptyDirectoryMarker: true})
 		return nil
 	}
+
+	var eg errgroup.Group
 	for _, f := range files {
```

Comment:
You can move the semaphore here.

Reply:
I don't think semaphore here does reasonable limit for parallelism as we call this function recursively?

Reply:
Ah I missed that. Then I guess a separate constructor would work.